src/main/java/com/cburch/logisim/circuit/appear/AppearanceSvgReader.java

```diff
@@ -132,18 +132,33 @@ private static DynamicElement getDynamicElement(String name, DynamicElement.Path
   }
 
   private static Boolean isInputPinReference(Element elt) {
-    final var width = Double.parseDouble(elt.getAttribute("width"));
-    final var radius = (int) Math.round(width / 2.0);
-    return AppearancePort.isInputAppearance(radius);
+    
+    if (elt.hasAttribute("dir")) {
+      final var direction = elt.getAttribute("dir");
+      return direction.equals("in");
+    } else {
```

Comment:
I'd probably drop `else` and corresponding brackets from `else` block.

Reply:
Done